test/hotspot/jtreg/compiler/c2/irTests/XorLNodeIdealizationTests.java

```diff
@@ -33,6 +33,11 @@
  * @run driver compiler.c2.irTests.XorLNodeIdealizationTests
  */
 public class XorLNodeIdealizationTests {
+
+    private static final long CONST_1 = RunInfo.getRandom().nextLong();
+    private static final long CONST_2 = RunInfo.getRandom().nextLong();
+    private static final long CONST_POW_2 = Math.abs(1L << RunInfo.getRandom().nextInt());
```

Comment:
Can you please use our new `Generators.java`? I'll explain why:

Sampling random powers-of-2, or values near powers-of-2 is very very rare with longs. There are about 64 powers-of-2 in a set of `2^64`, so the odds are really really bad. The sampler in `Generators.java` has a bias to pick more powers-of-2 and values nearby.

Reply:
And we need values nearby powers of 2, as your examples show ;)

Reply:
I'm a little hesitant to modify old tests. So I'd suggest to just do it for the new ones :smile: